ha/resource/dynamic_fid_service_ra.py

```diff
@@ -0,0 +1,267 @@
+# Foundation, either version 3 of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT ANY
+# WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+# PARTICULAR PURPOSE. See the GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License along
+# with this program. If not, see <https://www.gnu.org/licenses/>. For any questions
+# about this software or licensing, please email opensource@seagate.com or
+# cortx-questions@seagate.com.
+
+"""
+ ****************************************************************************
+ Description:       Generic resource agent used to map fid to cloneids.
+ ****************************************************************************
+"""
+
+import os
+import sys
+import time
+import pathlib
+
+from cortx.utils.log import Log
+
+sys.path.append(os.path.join(os.path.dirname(pathlib.Path(__file__)), '..', '..'))
+from ha.core.config.config_ha import ConfigHA
+from ha.resource.resource_agent import CortxServiceRA
+from ha.execute import SimpleCommand
+from ha.plugin.hare.hare import Hare
+from ha import const
+
+class FidManager:
+    """
+    Manage Fid for different services.
+    """
+
+    @staticmethod
+    def get_fid(service: str, node_id: str, instance_id: int) -> str:
+        """
+        Get Fid from hare mapping file for given services.
+
+        Args:
+            service ([str]): Service name.
+            node_id ([str]): Node name for fid instance.
+            instance_id ([int]): Instance id for service.
+
+        Returns:
+            str: Return fid of given service.
+        """
+        return Hare.get_fid(service, node_id, instance_id)
+
+class DynamicFidServiceRA(CortxServiceRA):
+    """
+    This class is used to provide wrapper around systemd resource agent.
+    This class manage fid mapping to (serviceName, NodeId, InstanceId).
+    Resource name: (Used to get instance)
+        like motr-ios-1, s3server-1, ...
+        Here, resource name is used to get instance.
+    Service name: (Used for systemd service)
+        like s3server, m0d, ...
+    Fid service name: (Used for service name in mapping)
+        like s3service, ios, confd
+    """
+    def __init__(self):
+        """
+        Initialize DynamicFidServiceRA class.
+        """
+        super(DynamicFidServiceRA, self).__init__()
+        self._execute = SimpleCommand()
+        self._status_list: list = ["failed", "active", "unknown"]
+
+    def _get_systemd_service(self) -> str:
+        """
+        Get Service name.
+
+        Returns:
+            str: Service name with fid mapping like service@fid
+        """
+        res_param = self.get_env()
+        service: str = res_param['OCF_RESKEY_service']
+        fid_service_name: str = res_param['OCF_RESKEY_fid_service_name']
+        # TODO: Get local node name from configuration
```

Comment:
We could use machine-id and then fetch node name from the conf store.

Reply:
Changed 
currently taking from pacemaker
 res_param["OCF_RESKEY_fid_service_name"]
avoiding from conf store as it will run in interval may add load to confstore